.github/workflows/test.yml

```diff
@@ -18,5 +19,20 @@ jobs:
           java-version: '11'
           distribution: 'adopt'
           cache: gradle
-      - name: Build with Gradle
-        run: ./gradlew build
+      - name: Cache SonarCloud packages
+        uses: actions/cache@v1
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+      - name: Cache Gradle packages
+        uses: actions/cache@v1
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
+          restore-keys: ${{ runner.os }}-gradle
+      - name: Build and analyze
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
```

Comment:
Where do these secrets live - I'm guessing they're set in the GitHub UI somewhere?
Could you add a comment describing what the GitHub token is called and which bot account owns it, just so it's easy to track down if it needs rotating at any point? Also perhaps something similar for the Sonar token?

Reply:
I'll add to team manual if that's ok?

Reply:
I'd be more keen to see it here, or at least a pointer to the team manual? If someone is looking at this in the future who isn't familiar with the page in the team manual they may not think to go looking there.

Reply:
@Wynndow I've removed the GITHUB_TOKEN with no discernable bad effect, and commented on the SONAR_TOKEN in the script

Reply:
https://media3.giphy.com/media/d6nvFKK6Dr9YbBDeNQ/giphy.gif